### PR TITLE
fix(deps): require six>=1.4.0

### PR DESCRIPTION
**Pull request by busunkim96**
fix(deps): add lower bound for `six`
test: use constraints file to test lower bounds

### CODE REVIEW

File: testing/constraints-3.6.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# This constraints file is used to check that lower bounds
+# are correct in setup.py
+# List *all* library dependencies and extras in this file.
+# Pin the version to the lower bound.
+#
+# e.g., if setup.py has "foo >= 1.14.0, < 2.0.0dev",
+# Then this file should have foo==1.14.0
+six==100000.0.0
```
Comment (busunkim96) [inline testing/constraints-3.6.txt:8]: Needs to be fixed.

File: setup.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
 
 REQUIREMENTS = [
-    'six',
+    'six>=1.4.0',
```
Comment (busunkim96) [inline setup.py:27]: From experimentation this was the lowest version that resulted in passing unit tests.

1.3.0 results in the following error:
```
_________________________________________ ERROR collecting tests_async/unit/requests/test_upload.py __________________________________________
Traceback (most recent call last):
  File "/usr/local/google/home/busunkim/github/google-resumable-media-python/tests_async/unit/requests/test_upload.py", line 23, in <module>
    import google._async_resumable_media.requests.upload as upload_mod
  File "/usr/local/google/home/busunkim/github/google-resumable-media-python/google/_async_resumable_media/requests/__init__.py", line 661, in <module>
    from google._async_resumable_media.requests.download import ChunkedDownload
  File "/usr/local/google/home/busunkim/github/google-resumable-media-python/google/_async_resumable_media/requests/download.py", line 21, in <module>
    from google._async_resumable_media.requests import _request_helpers
  File "/usr/local/google/home/busunkim/github/google-resumable-media-python/google/_async_resumable_media/requests/_request_helpers.py", line 26, in <module>
    import google.auth.transport._aiohttp_requests as aiohttp_requests
  File "/usr/local/google/home/busunkim/github/google-resumable-media-python/.nox/unit-3-6/lib/python3.6/site-packages/google/auth/__init__.py", line 19, in <module>
    from google.auth._default import default, load_credentials_from_file
  File "/usr/local/google/home/busunkim/github/google-resumable-media-python/.nox/unit-3-6/lib/python3.6/site-packages/google/auth/_default.py", line 30, in <module>
    import google.auth.transport._http_client
  File "/usr/local/google/home/busunkim/github/google-resumable-media-python/.nox/unit-3-6/lib/python3.6/site-packages/google/auth/transport/__init__.py", line 41, in <module>
    @six.add_metaclass(abc.ABCMeta)
AttributeError: module 'six' has no attribute 'add_metaclass'
```

`add_metaclass` was added in 1.4.0 https://github.com/benjaminp/six/blob/42636b15dd1a5b85de56eac98e47954d4c776576/CHANGES#L248-L249

**Comment by parthea** [comment]
@googlebot I consent.

**Review by tseaver — APPROVED** [review]

**Comment by tseaver** [comment]
Of course, we are going to be ripping `six` out altogether soon (fingers crossed, #229).

**Pull request merged**